### PR TITLE
Add options to specify categoryField + delimiter for FacetValueSuggestions

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/JSUI-2578

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/FacetValueSuggestionsTest.ts [hunk 1]
```diff
@@ -130,6 +130,78 @@ export function FacetValueSuggestionsTest() {
...
+      });
+
+      describe('with standard delimiter', () => {
+        beforeEach(() => {
+          setUpSuggestionsFromProviderToReturn([
+            { field: someField, keyword: aKeyword, numberOfResults: 10, score: { distanceFromTotalForField: 100 }, value: 'a|b|c|d' }
+          ]);
+        });
+
+        it('should modify the state by correctly splitting the field value', async done => {
```
Comment (samisayegh) [inline unitTests/ui/FacetValueSuggestionsTest.ts]: How is this test different from the one on line 143?

File: unitTests/ui/FacetValueSuggestionsTest.ts [hunk 2]
```diff
@@ -130,6 +130,78 @@ export function FacetValueSuggestionsTest() {
       });
     });
 
+    describe('when the field is a standard category field', () => {
```
[the remark this answers is not in the document]
Comment (olamothe) [reply unitTests/ui/FacetValueSuggestionsTest.ts]: Yeah that's a bad copy paste, fixing...

File: unitTests/ui/FacetValueSuggestionsTest.ts (hunk 1 above)
Comment (olamothe) [reply unitTests/ui/FacetValueSuggestionsTest.ts]: Should be fixed after removing bad copy + paste :)

File: src/ui/FacetValueSuggestions/FacetValueSuggestions.ts [hunk 3]
```diff
@@ -131,7 +133,25 @@ export class FacetValueSuggestions extends Component {
      */
     templateHelper: ComponentOptions.buildCustomOption<(row: IFacetValueSuggestionRow, omnibox: Omnibox) => string>(() => {
       return null;
-    })
+    }),
+    /**
+     * Specifies if the field to suggest is a "Category field", similar to the format used by the [`CategoryFacet`]{@link CategoryFacet} component.
```
Comment (jpmarceau) [inline src/ui/FacetValueSuggestions/FacetValueSuggestions.ts]: small suggestion:
if -> whether
Also, is the format merely 'similar' to that used by the category facet, or is it exactly the same? If it's the same, I'd go with:
..., similar to the format used by the ... -> ..., i.e., it has the format used by the ...
If it's just similar, let me know where I can get some details and I'll try to suggest something :)

Comment (olamothe) [reply src/ui/FacetValueSuggestions/FacetValueSuggestions.ts]: No you are right, it's exactly the same format.

File: unitTests/ui/FacetValueSuggestionsTest.ts [hunk 4]
```diff
@@ -130,6 +130,78 @@ export function FacetValueSuggestionsTest() {
...
+    });
+
+    describe('when the field is a category field', () => {
+      beforeEach(() => {
+        setUpKeywordInOmnibox(aKeyword);
+        setUpOmniboxSuggestionsToReturn([getOmniboxSuggestionValue()]);
+        test.cmp.options.isCategoryField = true;
+      });
+
+      describe('with standard delimiter', () => {
```
[the remark this answers is not in the document]
Comment (olamothe) [reply unitTests/ui/FacetValueSuggestionsTest.ts]: 🤷‍♂ Debatable, but I think in this case it makes things a bit easier to read with different `beforeEach` for each test.

**Pull request merged**